cli/src/pcluster/models/cluster.py

```diff
@@ -490,11 +491,44 @@ def _upload_artifacts(self):
                     PCLUSTER_S3_ARTIFACTS_DICT.get("instance_types_data_name"),
                     format=S3FileFormat.JSON,
                 )
+
+            if isinstance(self.config.scheduling, ByosScheduling):
+                self._render_and_upload_byos_template()
+        except BadRequestClusterActionError:
+            raise
         except Exception as e:
             message = f"Unable to upload cluster resources to the S3 bucket {self.bucket.name} due to exception: {e}"
             LOGGER.error(message)
             raise _cluster_error_mapper(e, message)
 
+    def _render_and_upload_byos_template(self):
```

Comment:
_upload_artifacts() is called after CDKTemplateBuilder, if upload failed, there is a invalid link for byos substack, will it affect the cluster creation?